runelite-client/src/main/java/net/runelite/client/chat/CommandManager.java

```diff
@@ -79,4 +97,39 @@ private void scriptEvent(ScriptCallbackEvent event)
 		CommandExecuted commandExecuted = new CommandExecuted(command, args);
 		eventBus.post(commandExecuted);
 	}
+
+	private void handleInput(ScriptCallbackEvent event)
+	{
+		Client client = clientProvider.get();
+		final String[] stringStack = client.getStringStack();
+		final int[] intStack = client.getIntStack();
+		int stringStackCount = client.getStringStackSize();
+		int intStackCount = client.getIntStackSize();
+
+		final String typedText = stringStack[stringStackCount - 1];
+		final int chatType = intStack[intStackCount - 1];
+
+		ChatboxInput chatboxInput = new ChatboxInput(typedText, chatType);
+		eventBus.post(chatboxInput);
+
+		if (chatboxInput.isStop())
+		{
+			// input was blocked.
+			stringStack[stringStackCount - 1] = ""; // prevent script from sending
+		}
+	}
+
+	public void sendChatboxInput(int chatType, String input)
```

Comment:
I would probably make this something like continueWithMessage so we will not have public API for sending messages to server